app/scripts/modules/amazon/serverGroup/details/serverGroupDetails.aws.controller.js

```diff
@@ -418,5 +425,46 @@ module.exports = angular.module('spinnaker.serverGroup.details.aws.controller',
         serverGroup.accountDetails = details;
       });
     };
+
+    this.lookForPipelineCommitDiffs = (executionId) => {
+      if (executionId) {
+        executionService.getExecution(executionId).then(executionDetails => {
+
+          this.jarDiffs = {
+            added: [],
+            downgraded: [],
+            duplicates: [],
+            removed: [],
+            unchanged: [],
+            unknown: [],
+            upgraded: []
+          };
+          this.commits = [];
+
+          const deployStages = executionDetails.stages.filter(stage => stage.type === 'deploy');
```

Comment:
Yeah, you should have `stageId` - we use it to create the source link. Not sure if there are cases where there would be multiple deploy stages deploying different artifacts, but it would be safer to pick the deploy stage from the tag.